apps/music/js/ui/views/player_view.js

```diff
@@ -1095,7 +1095,8 @@ var PlayerView = {
         // If we're within 1 second of the end of the song, register
         // a timeout to skip to the next song one second after the song ends
         if (this.audio.currentTime >= this.audio.duration - 1 &&
-            this.endedTimer == null) {
+            this.endedTimer == null &&
+            this.playStatus !== PLAYSTATUS_PAUSED) {
           var timeToNext = (this.audio.duration - this.audio.currentTime + 1);
           this.endedTimer = setTimeout(function() {
```

Comment:
Dave Hylands had theorized that there is/was a race condition that was biting us. I'm trying a few different build settings for B2G to see if I can reproduce the issue. If I can't, I think we should remove the workaround and then get people to use/test the music app to make sure everything is ok. With 3.0 being quite a ways in the future, I think now's the best time to do something like this.